### PR TITLE
Ethantr/feature/edit unavailability

### PR DESCRIPTION
**Pull request by ethantr**
## Describe your changes
- View and view model, miscellaneous for edit functionality
- Can prefill form with existing event information
- can send event with edited info 

## Issue ticket number and link
FE-72 https://fireapp-emergiq-2024.atlassian.net/jira/software/projects/FE/boards/1?selectedIssue=FE-72

### CODE REVIEW

File: fireapp/lib/l10n/app_en.arb [hunk 1]
```diff
@@ -132,9 +132,16 @@
   "enterStartTime" : "Schedule Start Time",
   "addSchedule" : "Add schedule",
 
+"calendarItemDelete": "Delete",
```
Comment (BenMMcLean) [inline fireapp/lib/l10n/app_en.arb]: Please correct indentation

File: fireapp/lib/presentation/unavailability_form/unavailability_form_widget.dart [hunk 2]
```diff
@@ -67,125 +48,142 @@ class _UnavailabilityFormState extends FireAppState<UnavailabilityForm>
...
+                                                ?.color ??
+                                            Colors.white)
+                                    ),
+                                  )
+                              ],
+                            ),
+                          );
+                        },
+                      );
+                    })
```
Comment (emilymclean) [inline fireapp/lib/presentation/unavailability_form/unavailability_form_widget.dart:103]: The indentation for this code is strange and makes it a little hard to read, please fix

File: fireapp/lib/presentation/unavailability_form/unavailability_form_widget.dart [hunk 3]
```diff
@@ -67,125 +48,142 @@ class _UnavailabilityFormState extends FireAppState<UnavailabilityForm>
...
+                            currentValue: value,
+                            onValueChanged: viewModel.updateEndDate,
+                            decoration:
+                                textFieldStylePositioned(context).copyWith(
+                              prefixIcon: const Icon(Icons.calendar_today),
+                              labelText: AppLocalizations.of(context)
+                                  ?.enterUnavailabilityEndDate,
+                            )
+                        );
+                      }),
```
Comment (emilymclean) [inline fireapp/lib/presentation/unavailability_form/unavailability_form_widget.dart:167]: Same thing with brackets

**Review by emilymclean — CHANGES_REQUESTED** [review]
Good work, just make some minor formatting changes

File: fireapp/lib/presentation/unavailability_form/unavailability_form_widget.dart (hunk 3 above)
Comment (ethantr) [reply fireapp/lib/presentation/unavailability_form/unavailability_form_widget.dart:167]: I'm having problems with dart format and my IDE restricting some of the indents, making it impossible to have it in the way you're asking, not sure how to go around this.

Comment (emilymclean) [reply fireapp/lib/presentation/unavailability_form/unavailability_form_widget.dart:167]: Damn dart is annoying. I'll just approve

**Review by emilymclean — APPROVED** [review]

**Pull request merged**